### PR TITLE
Migrate to GitHub Actions

### PR DESCRIPTION
**Pull request by dnerini**
Beginning of November 2020, Travis CI introduced a [new pricing model](https://blog.travis-ci.com/2020-11-02-travis-ci-new-billing) which in practice stopped providing unlimited hours to open source projects. In only few weeks, pysteps ran out of its allocated hours (which are not renewable). For this reason, we have decided to migrate to GitHub Actions, since this still offers unlimited hours to public repositories.

In short:
- test for linux-latest, macos-latest and windows-latest (note that the latter was not tested under Travis CI)
- test for Python 3.6 and 3.8 (the compromise strategy is to test for the oldest and latest supported python versions only)
- Black check runs on a separate workflow 

TODO/questions:
- [x] code coverage on Codecov
~~- [ ] should we add a pylint workflow?~~

### CODE REVIEW

File: .github/workflows/test_pysteps.yml [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+          python -c "import pysteps; print(pysteps.__file__)"
+          python -c "from pysteps import motion"
+          python -c "from pysteps.motion import vet"
+          python -c "from pysteps.motion import proesmans"
+
+      - name: Run tests
+        env:
+          PYSTEPSRC: ${{github.workspace}}/pysteps_data/pystepsrc
+        run: |
+          python -c "import pysteps; print(pysteps.__file__)"
```
Comment (dnerini) [inline .github/workflows/test_pysteps.yml]: Why retesting for module imports in this step?

**Comment by dnerini** [comment]
The black test is failing, but I suggest we reformat the code separately once all the remaining pull requests are merged, right before releasing v1.4,

**Comment by dnerini** [comment]
Did the tests fail, when you tried python 3.9?

File: .github/workflows/test_pysteps.yml [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+jobs:
+  unit_tests:
+    name: Unit Tests (${{ matrix.python-version }}, ${{ matrix.os }})
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ "ubuntu-latest"] #, "macos-latest", "windows-latest" ]
+        python-version: [ "3.6"] #, "3.8"]
```
Comment (RubenImhoff) [inline .github/workflows/test_pysteps.yml]: Why Python 3.6 and not 3.8? It looks like were testing against version 3.8, aren't we? But I may be wrong. ;)

File: .github/workflows/test_pysteps.yml [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+        env:
+          OS: ${{ matrix.os }}
+          PYTHON: ${{ matrix.python-version }}
+        with:
+          file: coverage.xml
+          directory: ${{github.workspace}}/pysteps_data
+          flags: unit_tests
+          env_vars: OS,PYTHON
+          fail_ci_if_error: true
+          verbose: true
```
Comment (RubenImhoff) [inline .github/workflows/test_pysteps.yml]: Github suggests here to have an empty newline to finalize the script. I don't know if it's necessary (not per se for Python at least).

**Review by RubenImhoff — COMMENTED** [review]
Great work on this short notice, @dnerini and @aperezhortal! Looks like we only have to make some code adjustments (later) to successfully pass the Black test, right?

File: .github/workflows/test_pysteps.yml (hunk 2 above)
Comment (dnerini) [reply .github/workflows/test_pysteps.yml]: I just commented 3.8 out for debugging, now it's back with ad1cb2b

File: .github/workflows/test_pysteps.yml (hunk 1 above)
Comment (aperezhortal) [reply .github/workflows/test_pysteps.yml]: oops, I think I was trying something and I forgot to remove that part.

**Comment by aperezhortal** [comment]
> Did the tests fail, when you tried python 3.9?

For py39 there were conflicts during the creation of the conda environment and the tests failed. I think that maybe we should wait a little bit until the packages dependencies are updated in conda-forge for the windows build. 

Once we can build the conda environment in windows, we could replace 3.8 with 3.9. In that way we test the minimum required python version and the latest version.

**Comment by aperezhortal** [comment]
> Great work on this short notice, @dnerini and @aperezhortal! Looks like we only have to make some code adjustments (later) to successfully pass the Black test, right?

The Black tests were added to check that the code base is compliant with the latest black version. Black is changing quite frequently so this can be used to indicate that we need to update black in our local development environment. 
Once we merge this PR, we can run black and push the changes to the main branch directly.

**Comment by dnerini** [comment]
> 
> 
> > Did the tests fail, when you tried python 3.9?
> 
> For py39 there were conflicts during the creation of the conda environment and the tests failed. I think that maybe we should wait a little bit until the packages dependencies are updated in conda-forge for the windows build.
> 
> Once we can build the conda environment in windows, we could replace 3.8 with 3.9. In that way we test the minimum required python version and the latest version.

Agreed. Btw, we have the same issue with the PR on our conda-forge pysteps-feedstock: https://github.com/conda-forge/pysteps-feedstock/pull/11

**Comment by aperezhortal** [comment]
> should we add a pylint workflow?

I was thinking about this too. Now we have several pep8 on the codebase and pylint gives a very detailed report on the code (that means, it will point out several violations). I think that we can postpone the pylint checks for a little while and perhaps start with some flake8 checks on future PRs. Adding and fixing one or a few types of errors at each PR.

Once the codebase passes many flake8 tests, we can certainly add pylint.

**Review by RubenImhoff — APPROVED** [review]

**Review by aperezhortal — APPROVED** [review]
